diff --git a/samples/extensions/workbench/Elsa.Server.Web/Program.cs b/samples/extensions/workbench/Elsa.Server.Web/Program.cs
--- a/samples/extensions/workbench/Elsa.Server.Web/Program.cs
+++ b/samples/extensions/workbench/Elsa.Server.Web/Program.cs
@@ -30,9 +30,10 @@
 using Elsa.Persistence.MongoDb.Modules.Tenants;
 using Elsa.Retention.Extensions;
 using Elsa.Retention.Models;
-using Elsa.Secrets.Extensions;
-using Elsa.Secrets.Management.Tasks;
-using Elsa.Secrets.Persistence;
+using Elsa.Secrets.Persistence.EFCore.Extensions;
+using Elsa.Secrets.Persistence.EFCore.PostgreSql.Extensions;
+using Elsa.Secrets.Persistence.EFCore.Sqlite.Extensions;
+using Elsa.Secrets.Persistence.EFCore.SqlServer.Extensions;
 using Elsa.Server.Web;
 using Elsa.Server.Web.Extensions;
 using Elsa.Server.Web.Filters;
@@ -93,7 +93,6 @@ const DistributedCachingTransport distributedCachingTransport = DistributedCachingTransport.MassTransit;
 const MassTransitBroker massTransitBroker = MassTransitBroker.Memory;
 const bool useMultitenancy = false;
 const bool useTenantsFromConfiguration = true;
-const bool useSecrets = false;
 const bool disableVariableWrappers = false;
 
 ObjectConverter.StrictMode = true;
@@ -101,6 +100,7 @@ ObjectConverter.StrictMode = true;
 var builder = WebApplication.CreateBuilder(args);
 var services = builder.Services;
 var configuration = builder.Configuration;
+var useSecrets = configuration.GetValue("Features:Secrets:Enabled", false);
 var identitySection = configuration.GetSection("Identity");
 var identityTokenSection = identitySection.GetSection("Tokens");
 var sqliteConnectionString = configuration.GetConnectionString("Sqlite")!;
@@ -619,32 +620,27 @@
         if (useSecrets)
         {
             elsa
-                .UseSecrets()
-                .UseSecretsManagement(management =>
-                {
-                    management.ConfigureOptions(options => configuration.GetSection("Secrets:Management").Bind(options));
-                    if (sqlDatabaseProvider == SqlDatabaseProvider.SqlServer)
-                        management.UseEntityFrameworkCore(ef =>
+                .UseSecrets(secrets =>
+                {
+                    secrets.ConfigureOptions = options => configuration.GetSection("Secrets").Bind(options);
+                    secrets.UseEntityFrameworkCore(ef =>
+                    {
+                        ef.UseContextPooling = useDbContextPooling;
+                        if (sqlDatabaseProvider == SqlDatabaseProvider.SqlServer)
                         {
-                            ef.UseContextPooling = useDbContextPooling;
                             ef.UseSqlServer(sqlServerConnectionString);
-                        });
-                    else if (sqlDatabaseProvider == SqlDatabaseProvider.PostgreSql)
-                        management.UseEntityFrameworkCore(ef =>
+                        }
+                        else if (sqlDatabaseProvider == SqlDatabaseProvider.PostgreSql)
                         {
-                            ef.UseContextPooling = useDbContextPooling;
                             ef.UsePostgreSql(postgresConnectionString);
-                        });
-                    else
-                        management.UseEntityFrameworkCore(ef =>
+                        }
+                        else
                         {
-                            ef.UseContextPooling = useDbContextPooling;
                             ef.UseSqlite(sp => sp.GetSqliteConnectionString());
-                        });
+                        }
+                    });
                 })
-                .UseSecretsApi()
-                .UseSecretsScripting()
-                ;
+                .UseSecretsJavaScript();
         }
 
         elsa.UseRetention(r =>
@@ -737,7 +733,6 @@
 {
     options.Schedule.ConfigureTask<TriggerBookmarkQueueRecurringTask>(TimeSpan.FromSeconds(300));
     options.Schedule.ConfigureTask<PurgeBookmarkQueueRecurringTask>(TimeSpan.FromSeconds(300));
-    options.Schedule.ConfigureTask<UpdateExpiredSecretsRecurringTask>(TimeSpan.FromHours(4));
     options.Schedule.ConfigureTask<RestartInterruptedWorkflowsTask>(TimeSpan.FromSeconds(15));
 });
 
diff --git a/samples/extensions/workbench/Elsa.Server.Web/Elsa.Server.Web.csproj b/samples/extensions/workbench/Elsa.Server.Web/Elsa.Server.Web.csproj
--- a/samples/extensions/workbench/Elsa.Server.Web/Elsa.Server.Web.csproj
+++ b/samples/extensions/workbench/Elsa.Server.Web/Elsa.Server.Web.csproj
@@ -75,12 +75,12 @@
 	  <ProjectReference Include="../../../../src/extensions/scheduling/Elsa.Scheduling.Quartz.EFCore.PostgreSql/Elsa.Scheduling.Quartz.EFCore.PostgreSql.csproj" />
 	  <ProjectReference Include="../../../../src/extensions/scheduling/Elsa.Scheduling.Quartz.EFCore.Sqlite/Elsa.Scheduling.Quartz.EFCore.Sqlite.csproj" />
 	  <ProjectReference Include="../../../../src/extensions/scheduling/Elsa.Scheduling.Quartz.EFCore.SqlServer/Elsa.Scheduling.Quartz.EFCore.SqlServer.csproj" />
-	  <ProjectReference Include="../../../../src/extensions/secrets/Elsa.Secrets.Api/Elsa.Secrets.Api.csproj" />
+	  <ProjectReference Include="../../../../src/modules/Elsa.Secrets/Elsa.Secrets.csproj" />
 	  <ProjectReference Include="../../../../src/modules/Elsa.Secrets.Persistence.EFCore.PostgreSql/Elsa.Secrets.Persistence.EFCore.PostgreSql.csproj" />
 	  <ProjectReference Include="../../../../src/modules/Elsa.Secrets.Persistence.EFCore.Sqlite/Elsa.Secrets.Persistence.EFCore.Sqlite.csproj" />
 	  <ProjectReference Include="../../../../src/modules/Elsa.Secrets.Persistence.EFCore.SqlServer/Elsa.Secrets.Persistence.EFCore.SqlServer.csproj" />
 	  <ProjectReference Include="../../../../src/modules/Elsa.Secrets.Persistence.EFCore/Elsa.Secrets.Persistence.EFCore.csproj" />
-	  <ProjectReference Include="../../../../src/extensions/secrets/Elsa.Secrets.Scripting/Elsa.Secrets.Scripting.csproj" />
+	  <ProjectReference Include="../../../../src/modules/Elsa.Secrets.JavaScript/Elsa.Secrets.JavaScript.csproj" />
 	  <ProjectReference Include="../../../../src/extensions/servicebus/Elsa.ServiceBus.Kafka/Elsa.ServiceBus.Kafka.csproj" />
 	  <ProjectReference Include="../../../../src/extensions/servicebus/Elsa.ServiceBus.MassTransit.AzureServiceBus/Elsa.ServiceBus.MassTransit.AzureServiceBus.csproj" />
 	  <ProjectReference Include="../../../../src/extensions/servicebus/Elsa.ServiceBus.MassTransit.RabbitMq/Elsa.ServiceBus.MassTransit.RabbitMq.csproj" />
